driver/src/main/java/org/neo4j/driver/internal/SimpleResult.java

```diff
@@ -76,7 +77,24 @@ public Value get( int fieldIndex )
     @Override
     public Value get( String fieldName )
     {
-        return current.get( fieldName );
+        if( current == null )
+        {
+            throw new ClientException(
+                    "The current record is null. " +
+                    "It is either because the result contains no more record, or the record has not been set. " +
+                    "You could call `next()` to test if there are more records in this result " +
+                    "and/or to iterate over result records." );
```

Comment:
I actually think it'd be enough to explain the case of not having called `next` - this will only happen if the user did not call `next`, or did not check the result of `next` being true. If you've reached the end of the stream, the result will point to the last record (assuming there was at least one record).

Eg. maybe this can be altered to highlight that scenario more - `In order to access fields of a record in a result, you must first call next() to point the result to the next record in the result stream.`